File: frontend/src/types/validation.ts
export type ValidationRequestStatus =
  | 'PENDING'
  | 'UNDER_REVIEW'
  | 'CONFIRMED'
  | 'REJECTED'
  | 'UNCERTAIN'
  | 'LAB_REFERRAL';

export type ValidationPriority = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export type LabReferralStatus =
  | 'REQUESTED'
  | 'SAMPLE_COLLECTED'
  | 'IN_LAB'
  | 'RESULT_AVAILABLE'
  | 'CLOSED';

export interface ExpertValidationRecord {
  id: string;
  validation_request_id: string;
  expert_user_id: string;
  expert_name?: string;
  status: ValidationRequestStatus;
  confirmed_condition?: string;
  expert_notes?: string;
  farmer_guidance?: string;
  rejection_reason?: string;
  uncertain_recommendation?: string;
  validation_version: string;
  validated_at: string;
}

export interface LabReferral {
  id: string;
  referral_code: string;
  validation_request_id: string;
  farm_id: string;
  zone_id?: string;
  sample_type: string;
  suspected_condition: string;
  reason: string;
  status: LabReferralStatus;
  requested_at: string;
  result_summary?: string;
  result_date?: string;
}

export interface ExpertValidationRequest {
  id: string;
  case_number: string;
  analysis_id?: string;
  farm_id: string;
  farm_name?: string;
  zone_id?: string;
  zone_name?: string;
  crop_id?: string;
  crop_name?: string;
  requested_by: string;
  requester_name?: string;
  assigned_expert_id?: string;
  assigned_expert_name?: string;
  priority: ValidationPriority;
  status: ValidationRequestStatus;
  reason: string;
  suspected_condition: string;
  ai_confidence: number;
  crop_growth_stage?: string;
  symptoms: string[];
  image_urls: string[];
  weather_summary?: {
    temperature_c?: number;
    humidity_pct?: number;
    rainfall_mm?: number;
    disease_risk_level?: string;
  };
  hotspot_id?: string;
  drone_observation_id?: string;
  due_at?: string;
  completed_at?: string;
  created_at: string;
  updated_at?: string;
  decisions?: ExpertValidationRecord[];
  lab_referrals?: LabReferral[];
}

export interface ValidationStats {
  total_pending: number;
  high_priority: number;
  critical: number;
  my_assigned: number;
  recently_validated: number;
  lab_referrals: number;
}

File: backend/app/schemas/validation.py
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field, ConfigDict

from app.models.validation import (
    ValidationRequestStatus,
    ValidationPriority,
)
from app.models.laboratory import LabReferralStatus


class ValidationRequestCreate(BaseModel):
    analysis_id: Optional[str] = None
    farm_id: str
    zone_id: Optional[str] = None
    crop_id: Optional[str] = None
    priority: ValidationPriority = ValidationPriority.MEDIUM
    reason: str
    suspected_condition: str = "Suspected Foliar Anomaly"
    ai_confidence: float = Field(default=0.75, ge=0.0, le=1.0)
    crop_growth_stage: Optional[str] = None
    symptoms: List[str] = Field(default_factory=list)
    image_urls: List[str] = Field(default_factory=list)
    weather_summary: Dict[str, Any] = Field(default_factory=dict)
    hotspot_id: Optional[str] = None
    drone_observation_id: Optional[str] = None


class ValidationDecisionSubmit(BaseModel):
    decision: ValidationRequestStatus = Field(
        ..., description="CONFIRMED, REJECTED, UNCERTAIN, or LAB_REFERRAL"
    )
    confirmed_condition: Optional[str] = None
    expert_notes: Optional[str] = Field(None, description="Internal technical notes for archive")
    farmer_guidance: Optional[str] = Field(None, description="Plain language actionable guidance for farmer")
    rejection_reason: Optional[str] = None
    uncertain_recommendation: Optional[str] = None


class LabReferralCreate(BaseModel):
    sample_type: str = "Leaf Tissue Sample"
    suspected_condition: str
    reason: str


class LabReferralResponse(BaseModel):
    id: str
    referral_code: str
    validation_request_id: str
    farm_id: str
    zone_id: Optional[str] = None
    sample_type: str
    suspected_condition: str
    reason: str
    status: LabReferralStatus
    requested_at: datetime
    result_summary: Optional[str] = None
    result_date: Optional[datetime] = None

    class Config:
        from_attributes = True


class ExpertValidationRecordResponse(BaseModel):
    id: str
    validation_request_id: str
    expert_user_id: str
    expert_name: Optional[str] = None
    status: ValidationRequestStatus
    confirmed_condition: Optional[str] = None
    expert_notes: Optional[str] = None
    farmer_guidance: Optional[str] = None
    rejection_reason: Optional[str] = None
    uncertain_recommendation: Optional[str] = None
    validation_version: str
    validated_at: datetime

    class Config:
        from_attributes = True


class ValidationRequestResponse(BaseModel):
    id: str
    case_number: str
    analysis_id: Optional[str] = None
    farm_id: str
    farm_name: Optional[str] = None
    zone_id: Optional[str] = None
    zone_name: Optional[str] = None
    crop_id: Optional[str] = None
    crop_name: Optional[str] = None
    requested_by: str
    requester_name: Optional[str] = None
    assigned_expert_id: Optional[str] = None
    assigned_expert_name: Optional[str] = None
    priority: ValidationPriority
    status: ValidationRequestStatus
    reason: str
    suspected_condition: str
    ai_confidence: float
    crop_growth_stage: Optional[str] = None
    symptoms: List[str] = Field(default_factory=list)
    image_urls: List[str] = Field(default_factory=list)
    weather_summary: Dict[str, Any] = Field(default_factory=dict)
    hotspot_id: Optional[str] = None
    drone_observation_id: Optional[str] = None
    due_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    decisions: List[ExpertValidationRecordResponse] = Field(default_factory=list)
    lab_referrals: List[LabReferralResponse] = Field(default_factory=list)

    class Config:
        from_attributes = True


class ValidationStatsResponse(BaseModel):
    total_pending: int
    high_priority: int
    critical: int
    my_assigned: int
    recently_validated: int
    lab_referrals: int
